pkg/kubelet/cadvisor/rancher/client.go

```diff
@@ -274,7 +274,7 @@ func (self *Client) getEventStreamingData(url string, einfo chan *info.Event) er
 	for {
 		err := dec.Decode(m)
 		if err != nil {
-			if err == io.EOF {
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
UnexpectedEndOfFile is thrown when Rancher server goes down triggering rancher-agent restart. Cadvisor is not available till rancher comes up. Not sure if this is the right way to workaround the issue. As we are thinking about moving out cadvisor to its own microservice, the issue will be gone once it is done.